blacklight-cornell/app/controllers/catalog_controller.rb

```diff
@@ -44,7 +44,40 @@ def start_new_search_session?
     query_present || adv_query_present || filters_present
   end
 
-  #  DACCESS-215
+  # ============================================================================
+  # Looks for a matching saved search in history, and either returns it or
+  # creates a new Search. Any new Search is added to search history.
+  # ----------------------------------------------------------------------------
+  def find_or_initialize_search_session_from_params(params)
+    params_copy = normalize_saved_search_params(params)
```

Comment:
We'd talked about how SearchHistoryHelper#build_search_history_url adds extra params to the url that the original search didn't have, e.g. a default sort. If we made sure to only generate a search url from existing params, do we still need to override find_or_initialize_search_session_from_params from blacklight? It would be good to see how blacklight generates search history urls now - you may be able to just reuse or extend existing methods.

Reply:
I'll look into it 👍